test/e2e/util/util.go

```diff
@@ -396,3 +401,126 @@ func newClusterdeployment(clusterName string) *unstructured.Unstructured {
 		},
 	}
 }
+
+func newHostedcluster(hostedClusterNamespace, clusterName string) *unstructured.Unstructured {
```

Comment:
Over here, since this repo has a direct dependency on hypershift, we can use the structured client to do our test, which is easier for us to write up stuff. But if you don't have time, we can use `unstructure` client for now.
 
```
hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
	
hCluster := &hyperv1.HostedCluster{}
err := r.client.Get(ctx, request.NamespacedName, hCluster)
```
[^Code Ref](https://github.com/ianzhang366/managedcluster-import-controller/blob/bda99c120eff90dd2bb9c2753f21055987888fc8/pkg/controller/hypershift/hostedcluster_controller.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L98-L99)

Also, need to add the following:

```
	utilruntime.Must(hyperv1.AddToScheme(scheme))
```
[^Code ref](https://github.com/tphee/managedcluster-import-controller/blob/bda99c120eff90dd2bb9c2753f21055987888fc8/cmd/manager/main.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L66)

To the following to register the scheme to a structured client when initial the ginkgo test env,
https://github.com/ianzhang366/managedcluster-import-controller/blob/bda99c120eff90dd2bb9c2753f21055987888fc8/test/e2e/e2e_suite_test.go#L44

Reply:
Since it is working now and is consistent with the way Hive Cluster Deployment does the test, I think I will leave it as it is.